passive/ShipMuonShield.cxx

```diff
@@ -147,7 +147,7 @@ void ShipMuonShield::CreateArb8(TString arbName, TGeoMedium *medium,
   TGeoVolume *magF =
       gGeoManager->MakeArb8(arbName, medium, dZ, corners.data());
   magF->SetLineColor(color);
-  magF->SetField(magField);
+  if (fDesign != 11 || arbName.Contains("Absorb")){magF->SetField(magField);}

```

Comment:
As far as I remember built-in fields have a higher priority  so we will have 0-value field instead of FieldMap.

Reply:
That is very interesting. Because as I have mentioned in the link, this is how things are implemented for Hadron Stopper. If it is indeed the case, it means that hadron stopper will never have realistic field map but rather just 0 field. @ThomasRuf have you spotted anythign like that?
@PlaZMaD I am not experienced enough to know this detail for sure, can I just ask you to experiment and try to build it with setting field to 0 and realistc map. Then just check if fieldmap was applied?

Reply:
If you read what is written in geomGeant4, if there is a field map, this will take priority over const field of volume:

**    # Force the VMC to update/reset the fields defined by the fieldMaker object.
    # Get the ROOT/Geant4 geometry manager
     geom = ROOT.TG4GeometryManager.Instance()
    # Let the geometry know about the fieldMaker object
     geom.SetUserPostDetConstruction(fieldMaker)
    # Update the fields via the overriden ShipFieldMaker::Contruct() function
     geom.ConstructSDandField()

For this reason, we have all the if statements, only adding a field map in case const field of volume is not used.

Reply:
Looks like an old interface for "region" fields (like local ones, but are suppressed by global).
Hardcoded fields probably should be passed automatically if "/mcDet/setIsLocalMagField" is set "true" in g4config.in.

Reply:
> Looks like an old interface for "region" fields (like local ones, but are suppressed by global).
> Hardcoded fields probably should be passed automatically if "/mcDet/setIsLocalMagField" is set "true" in g4config.in.

Yes, from what I see [here](https://root.cern/magnetic-field), you are right. Please, see discussion below.